File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        freq = {}
        for i in range(1, 27):
            freq[chr(96 + i)] = 26 - i + 1

        total = 0
        for i in range(len(s)):
            total += (freq.get(s[i]) * (i + 1))

        return total

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var reverseDegree = function (s) {
    const freq = new Map();

    for (let i = 1; i <= 26; i++) {
        freq.set(String.fromCharCode(96 + i), 26 - i + 1);
    }

    let total = 0;
    for (let i = 0; i < s.length; i++) {
        total += (freq.get(s[i]) * (i + 1))
    }

    return total;
};